Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -308,6 +310,13 @@ protected TJob BuildJob(DirectoryInfoBase jobDirectory, bool nullJobOnError = tr
                 // Return a job representing an error for no runnable script file found for job
                 if (nullJobOnError)
                 {
+                    if (ex is IOException)
+                    {
+                        // unexpected IOException given simply reading files/folders.
+                        // just rethrow and let the upper layers handle it.
+                        throw;
+                    }
```

Comment:
This is the key fix.   This rethrows unexpected IOException and let the upper layers retry.